refactor: implement new compare logic (#32)
CeresDB/sqlness: README.md

```
It will do following things:
1. Collect all environments(first-level directory) under `basic-case`.
2. Run tests(`.sql` files) under environment one after one.

>    1. Before execution it will read `{testcase}.result`(create one if not exists) to `Vec` as compare base.

Reviewer: ```suggestion
   1. Before execution it will read `{testcase}.result`(create one if not exists) to memory for compare.
```

   2. During execution it will collect query response and write to `{testcase}.result`

>    3. After execution it will diff content of `Vec` with `{testcase}.result`, **PASS** when they are the same otherwise **FAIL**.

Reviewer: ```suggestion
   3. After execution it will compare the generated `{testcase}.result` with previous one, **PASS** when they are the same, and **FAIL** otherwise.
```

3. Report result.

Usually `result` files should be tracked in git, whenever there are failed tests, users should

> 1. Update `result` to latest version(e.g. `git add`) if result is right, or

Reviewer: ```suggestion
1. Update `result` to latest version(e.g. `git add`) if the newer result is right, or
```

> 2. Restore `result` back to original version(e.g. `git checkout`), troubleshoot bugs in database implementation, and run tests again

Reviewer: ```suggestion
2. Restore `result` back to original version (e.g. `git checkout`), troubleshoot bugs in database implementation, and run tests again
```
